onboard/nav/avoidances.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef AVOIDANCES_HPP
+#define AVOIDANCES_HPP
+
+#include "avoidance.hpp"
+
+// This class is the representation of different avoidance algorithms
+enum class AvoidanceType
+{
+    ORIGINAL,
+    UNKNOWN
+};
+
+Avoidance* AvoidFactory( StateMachine* stateMachine, AvoidanceType type );
+
+/*************************************************************************/
+/* Default Avoidance */
+/*************************************************************************/
+class Original : public Avoidance 
```

Comment:
Add comment for the class. This should be in the form `// This class ...` rather than `/*****/ comment /******/`